cpp/data/frameProcessor/src/XspressProcessPlugin.cpp

```diff
@@ -345,6 +347,26 @@ void XspressProcessPlugin::setup_memory_allocation()
     free(inp_est_memblock_);
   }
   inp_est_memblock_ = malloc(sizeof(double) * this->frames_per_block_ * num_channels_);
+
+  if (sum_memblock_)
+  {
+    free(sum_memblock_);
+  }
+  sum_memblock_ = malloc(sizeof(double) * this->frames_per_block_);
+  }
+
+void XspressProcessPlugin::calculate_sum(boost::shared_ptr<Frame> frame)
+{
+  double sum_value = 0;
+  const uint32_t *data = static_cast<const uint32_t *>(frame->get_image_ptr());
+  size_t elements_count = frame->get_image_size() / sizeof(uint32_t);
+  for (size_t pixel_index = 0; pixel_index < elements_count; pixel_index++)
+  {
+    sum_value += data[pixel_index];
+  }
+  double *sum_ptr = (double *)sum_memblock_;
+  sum_ptr += num_scalars_recorded_;
+  memcpy(sum_ptr, &sum_value, sizeof(double));
 }
```

Comment:
This needs a docstring explaining what it is doing. Is it summing `num_scalars_recorded_` values and then inserting that at the offset that would be the end of that block in the data array, leaving the rest blank? Why does it do this?

Reply:
So, **sum_ptr** is an array with the size equals to the number of frames in your block. What I was trying to do in that part of the code is use the **num_scalars_recorded_** variable to shift the position inside the array to which we were going to write the summed up data. So each frame is received and summed and then added to the corresponding position in the **sum_ptr** array.

Reply:
That would be good. I am still not sure I understand why you are shifting `sum_ptr` by `num_scalars_recorded` and not just `1`. Are you intentionally leaving gaps of unset values in `sum_memblock_` and if so why?

Reply:
Oh sorry! I am being silly. It resets to the start of the memory block each time this is called and then you set it to where you have gotten to each time. The first time I read this I was thinking it increments by `num_scalars_recorded_` every time.

OK that's fine so just add a docstring saying it calculates the sum for the frame and inserts the result into the next free position based on the number of scalars recorded so far, or something like that.

Reply:
Cool!
I added some explanation of what the method does!